### PR TITLE
SO Table Improvements on Status

### PR DESCRIPTION
**Pull request by WaggMe**
* improve filter, sort, search for SO and GSO tables

* add explorer resolver, and use in links for authority and staking options

* Chore: add DUAL for pkToAsset

* Chore: improve date handling for readability

### CODE REVIEW

**Comment by DonDuala** [comment]
Worth mimicking on GSO table too to stay consistent

**Comment by WaggMe** [comment]
> Worth mimicking on GSO table too to stay consistent

Definitely! Went for the quick win, adding to other tables now.

**Comment by WaggMe** [comment]
@DonDuala 🚢

File: app/src/pages/stakingOptions/stakingOptions.tsx [hunk 1]
```diff
@@ -166,20 +169,28 @@ export const StakingOptions = (props: { network: string }) => {
 
   const soFilters: Array<any> = [
     {
-      text: 'CSA',
-      value: 'csa',
+      text: 'Bonus',
```
Comment (DonDuala) [inline app/src/pages/stakingOptions/stakingOptions.tsx:172]: Let's replace Bonus & Integration with "MM" in both tables. So you can easily view the rewards sent to all MMs (market makers)

File: app/src/pages/stakingOptions/stakingOptions.tsx [hunk 2]
```diff
@@ -223,7 +232,9 @@ export const StakingOptions = (props: { network: string }) => {
       render: (authority) => {
         // TODO Make copyable
         return (
-          <a href={`${explorerStub}account/${authority.toBase58()}`}>{`${authority.toBase58().substring(0, 4)}...`}</a>
+          <a href={Config.explorerUrl(authority.toBase58())} target="_blank" rel="noreferrer">
```
Comment (DonDuala) [inline app/src/pages/stakingOptions/stakingOptions.tsx:235]: Open explorer links in a new tab

File: app/src/pages/stakingOptions/stakingOptions.tsx (hunk 1 above)
Comment (WaggMe) [reply app/src/pages/stakingOptions/stakingOptions.tsx]: Selecting 'MM' now filters for any of 'mm', 'bonus', 'integration'. Users can still pick just Bonus or Integration.

image

File: app/src/pages/stakingOptions/stakingOptions.tsx (hunk 2 above)
Comment (WaggMe) [reply app/src/pages/stakingOptions/stakingOptions.tsx:246]: This was already handled on authorities, added to the name link.

**Review by DonDuala — APPROVED** [review]

**Pull request merged**